room/integration-tests/testapp/src/androidTest/java/androidx/room/integration/testapp/test/TypeConverterFactoryTest.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2018 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.room.integration.testapp.test;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+
+import android.content.Context;
+
+import androidx.annotation.NonNull;
+import androidx.room.Database;
+import androidx.room.Room;
+import androidx.room.RoomDatabase;
+import androidx.room.TypeConverter;
+import androidx.room.TypeConverterFactory;
+import androidx.room.TypeConverters;
+import androidx.room.integration.testapp.TestDatabase;
+import androidx.room.integration.testapp.dao.PetDao;
+import androidx.room.integration.testapp.vo.Pet;
+import androidx.room.integration.testapp.vo.PetWithUser;
+import androidx.room.integration.testapp.vo.Toy;
+import androidx.room.integration.testapp.vo.User;
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.MediumTest;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.Date;
+
+@MediumTest
+@RunWith(AndroidJUnit4.class)
+public class TypeConverterFactoryTest {
+
+    @Test
+    public void testTypeConverterFactory() {
+        Context context = ApplicationProvider.getApplicationContext();
+        context.deleteDatabase("test");
+        TestDatabaseWithConverter db = null;
+        Throwable throwable = null;
+        try {
+            db = Room.databaseBuilder(context, TestDatabaseWithConverter.class, "test")
```

Comment:
Use the `inMemoryDatabaseBuilder` and then you can remove the codes regarding deleting the database file and it isolates the test more. Meanwhile, no need to surround `db` initialization with a try-catch, if something goes wrong the test will fail due to an exception opening or closing the DB, same as what you have now, but its just less noise in the test.